### ISSUE TITLE
Should https-proxy-agent be a dependency?

### ISSUE THREAD

**Issue by ryan-rushton** [auth0/node-jwks-rsa#184]
### Description

Upon install of my project I am seeing an error where https-proxy-agent isn't available for jwks-rsa. Should this be a hard dependency?

### Reproduction

Upon install and start I am receiving the following. Adding https-proxy-agent fixes the issue.

```
internal/modules/cjs/loader.js:1068
  throw err;
  ^

Error: Cannot find module 'https-proxy-agent'
Require stack:
- /Users/ryanrushton/dev/my-project/node_modules/jwks-rsa/lib/wrappers/request.js
- /Users/ryanrushton/dev/my-project/node_modules/jwks-rsa/lib/JwksClient.js
- /Users/ryanrushton/dev/my-project/node_modules/jwks-rsa/lib/index.js
- /Users/ryanrushton/dev/my-project/dist/src/common/auth/authentication.middleware.js
- /Users/ryanrushton/dev/my-project/dist/src/app.module.js
- /Users/ryanrushton/dev/my-project/dist/src/main.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:1065:15)
    at Function.Module._load (internal/modules/cjs/loader.js:911:27)
    at Module.require (internal/modules/cjs/loader.js:1125:19)
    at require (internal/modules/cjs/helpers.js:75:18)
    at Object.<anonymous> (/Users/ryanrushton/dev/my-project/node_modules/jwks-rsa/lib/wrappers/request.js:52:24)
    at Module._compile (internal/modules/cjs/loader.js:1236:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1257:10)
    at Module.load (internal/modules/cjs/loader.js:1085:32)
    at Function.Module._load (internal/modules/cjs/loader.js:950:14)
    at Module.require (internal/modules/cjs/loader.js:1125:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/ryanrushton/dev/my-project/node_modules/jwks-rsa/lib/wrappers/request.js',
    '/Users/ryanrushton/dev/my-project/node_modules/jwks-rsa/lib/JwksClient.js',
    '/Users/ryanrushton/dev/my-project/node_modules/jwks-rsa/lib/index.js',
    '/Users/ryanrushton/dev/my-project/dist/src/common/auth/authentication.middleware.js',
    '/Users/ryanrushton/dev/my-project/dist/src/app.module.js',
    '/Users/ryanrushton/dev/my-project/dist/src/main.js'
  ]
}
```

### Environment

> Please provide the following:

- **Version of this library used:**
^1.9.0
- **Version of the platform or framework used, if applicable:**
nestjs@^7.4.4 / express@^4.17.1
- **Other relevant versions (language, server software, OS, browser):**
- **Other modules/plugins/libraries that might be involved:**
Here is a list of my dependencies
```
"dependencies": {
    "@admin-bro/express": "^3.0.0",
    "@admin-bro/typeorm": "^1.3.0",
    "@nestjs/common": "^7.4.4",
    "@nestjs/config": "^0.5.0",
    "@nestjs/core": "^7.4.4",
    "@nestjs/platform-express": "^7.4.4",
    "@nestjs/typeorm": "^7.1.4",
    "admin-bro": "^3.2.5",
    "dotenv": "^8.2.0",
    "express": "^4.17.1",
    "express-formidable": "^1.2.0",
    "express-jwt": "^6.0.0",
    "express-session": "^1.17.1",
    "jwks-rsa": "^1.9.0",
    "passport": "^0.4.1",
    "passport-auth0": "^1.3.3",
    "pg": "^8.3.3",
    "reflect-metadata": "^0.1.13",
    "rimraf": "^3.0.2",
    "rxjs": "^6.6.3",
    "typeorm": "^0.2.26"
  },
```

**Comment by catchin** [comment]
Duplicate of #179

**Comment by ralphtheninja** [comment]
Just merge https://github.com/auth0/node-jwks-rsa/pull/180/files and be done with it.

**Comment by davidpatrick** [comment]
Fixed in #180
